google-cloud-firestore/src/main/java/com/google/cloud/firestore/Query.java

```diff
@@ -1758,6 +1763,103 @@ public ApiFuture<QuerySnapshot> get() {
     return get(null);
   }
 
+  ApiFuture<QueryProfile<QuerySnapshot>> getQueryProfileInfo(QueryMode queryMode) {
+    final SettableApiFuture<QueryProfile<QuerySnapshot>> result = SettableApiFuture.create();
+
+    RunQueryRequest.Builder request = RunQueryRequest.newBuilder();
+    request
+        .setStructuredQuery(buildQuery())
+        .setParent(options.getParentPath().toString())
+        .setMode(queryMode);
+
+    final List<QueryDocumentSnapshot> documentSnapshots = new ArrayList<>();
```

Comment:
Considering `readTime`, `queryPlan` and `statsStruct` are all defined within the anonymous `ResponseObserver` class, I think you should move `documentSnapshots` there as well.